src/YesSql.Provider.SqlServer/SqlServer2008R2Dialect.cs

```diff
@@ -0,0 +1,22 @@
+namespace YesSql.Provider.SqlServer
+{
+    public class SqlServer2008Dialect : SqlServerDialect
+    {
+        public override string Name => "SqlServer 2008 R2";
+
+        public override void Page(ISqlBuilder sqlBuilder, string offset, string limit)
+        {
+            if (limit != null)
+            {
+                var selector = sqlBuilder.GetSelector();
+                selector = " top " + limit + " " + selector;
+                sqlBuilder.Selector(selector);
+            }
+
+            if (offset != null)
+            {
+                sqlBuilder.WhereAnd("RowNum >" + offset);
```

Comment:
What is RownNum? I've tested a query using it like the one yessql would produce but it returns an error:
`SELECT DISTINCT TOP 10  [Document].*, [UserIndex].[NormalizedUserName] FROM [Document] INNER JOIN [UserIndex] ON [UserIndex].[DocumentId] = [Document].[Id] WHERE RowNum > 0 ORDER BY [UserIndex].[NormalizedUserName]`

This is the error I receive `'Msg 207, Level 16, State 1, Line 27
Invalid column name 'RowNum'`
I just ask cause I'm woriking on a new dialect supporting SqlServer2008 but the only way I found to do pagination produces a query like that:
SELECT * FROM (SELECT DISTINCT [Document].*, ROW_NUMBER() OVER( ORDER BY [UserIndex].[NormalizedUserName]) AS Seq,[UserIndex].[NormalizedUserName] FROM [Document] INNER JOIN [UserIndex] ON [UserIndex].[DocumentId] = [Document].[Id] ) yessqlQuery  WHERE Seq BETWEEN 1 AND 10

Exists a simpler way of doing it?

Reply:
My old Harddisk is corrupted I may need to install SQL Server 2008 R2 and try the unit tests again